OpenRA.Mods.Common/Activities/WaitForTurretAlignment.cs

```diff
@@ -0,0 +1,42 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Activities;
+using OpenRA.Mods.Common.Traits;
+
+namespace OpenRA.Mods.Common.Activities
+{
+    public class WaitForTurretAlignment : Activity
+    {
+        readonly Turreted turreted;
+        readonly int desiredAlignment;
+
+        public WaitForTurretAlignment(Actor self, int desiredAlignment)
+        {
+            this.turreted = self.Trait<Turreted>();
+            this.desiredAlignment = desiredAlignment;
+        }
+
+        protected override void OnFirstRun(Actor self)
+        {
+            turreted.StopAiming(self);
+            turreted.DesiredFacing = desiredAlignment;
+        }
+
+        public override Activity Tick(Actor self)
+        {
+            if (turreted.HasAchievedDesiredFacing)
```

Comment:
It looks like there are several places that could reset the `DesiredFacing` to something else (e.g. the `RealignDelay` expiring), which would break the waiting here.  We will need to check and re-set the desired facing if necessary.

Reply:
Alright I'll add something to handle this. I'm pretty busy ATM but I'll aim to implement this for the coming weekend.

Reply:
@pchote I made it so that tick sets DesiredFacing. I couldn't think of a faster way in terms of performance to do it without involving something like a mutex. I think if we added a conditional to check if it's changed it's going to be slower.